ot/bregman.py

```diff
@@ -919,8 +927,11 @@ def get_reg(n):  # exponential decreasing
     while loop:
 
         regi = get_reg(cpt)
-
-        G, logi = sinkhorn_stabilized(a, b, M, regi,
+        # b.flatten() is here for the case when 'sinkhorn_epsilon_scaling' has been called
+        # from sinkhorn2. In this casen b is a 2D array of size [..,1] and 'sinkhorn_stabilized'
+        # does not return a valid coupling, but rather transport cost.
+        # This is a patch to issue #221
+        G, logi = sinkhorn_stabilized(a, b.flatten(), M, regi,
```

Comment:
I was looking at the same issue earlier today, I don't think this is an ok way of doing this tbh. Flattening may create further bugs. I'm of the opinion of delegating the role of making sure the inputs are one dimensional to sinkhorn2 instead of implicitly converting them (several times) in the loop.

Reply:
Otherwise, I also had in mind that the transport plan sinkhorn functions could return batched transport plans if n_hists > 1, but this may require more work...